packages/typescript-eslint/src/config-helper.ts

```diff
@@ -91,14 +91,17 @@ export function config(
       return config;
     }
 
-    if (config.files) {
-      const files = config.files;
-      return [
-        ...extendsArr.map(conf => ({ ...conf, files: [...files] })),
-        config,
-      ];
-    }
+    const extension = {
+      ...(config.files && { files: config.files }),
+      ...(config.ignores && { ignores: config.ignores }),
```

Comment:
`ignores` will overwrite `ignores` from the extended config. I'm not sure if this is the desired behavior, but so far I've made it to match the behavior of `files`. Any thoughts on this?

Reply:
I think overriding is correct.
Any local config should always override the extended config